src/Validator/CnpValidator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Validator;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class CnpValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Cnp) {
+            throw new UnexpectedTypeException($constraint, Cnp::class);
+        }
+
+        $regexResponse = preg_match_all(
+            '/^([1-8])(\d{2})(0[1-9]|1[0-2])(0[1-9]|[1-2]\d|3[0-1])(0\d|[1-3]\d|4[0-8])(\d{3})(\d)$/',
+            $value,
+            $matches
+        );
```

Comment:
Counties 51 (Călărași) and 52(Giurgiu) are not in your regex expression... https://ro.wikipedia.org/wiki/Cod_numeric_personal_(Rom%C3%A2nia).... probably you have looked on another cnp description?
--


[Călărași](https://ro.wikipedia.org/wiki/Jude%C8%9Bul_C%C4%83l%C4%83ra%C8%99i) and 52